test/integration/framework/goleak.go

```diff
@@ -1,114 +0,0 @@
-		// created by: google.golang.org/grpc/internal/transport.(*serverHandlerTransport).HandleStreams
-		goleak.IgnoreTopFunction("sync.runtime_notifyListWait"),
-
-		// etcd
-		// created by: go.etcd.io/etcd/client/v3.(*watchGrpcStream).waitCancelSubstreams.func1
-		goleak.IgnoreTopFunction("go.etcd.io/etcd/client/v3.(*watchGrpcStream).waitCancelSubstreams.func1.1"),
-		// created by: go.etcd.io/etcd/client/v3.(*watchGrpcStream).newWatchClient
-		goleak.IgnoreTopFunction("go.etcd.io/etcd/client/v3.(*watchGrpcStream).serveSubstream.func1"),
-
-		// kcp / kube
-		// created by k8s.io/apiserver/pkg/registry/generic/registry.(*Store).startObservingCount
-		goleak.IgnoreTopFunction("k8s.io/apimachinery/pkg/util/wait.BackoffUntil"),
-		// created by: github.com/kcp-dev/kcp/pkg/informer.NewGenericDiscoveringDynamicSharedInformerFactory[...]
-		goleak.IgnoreTopFunction("github.com/kcp-dev/kcp/pkg/informer.NewGenericDiscoveringDynamicSharedInformerFactory[...].func3"),
-		// created by: k8s.io/apiserver/pkg/storage/storagebackend/factory.newETCD3Check
-		goleak.IgnoreTopFunction("k8s.io/apiserver/pkg/storage/storagebackend/factory.newETCD3Check.func2"),
-
-		// Known from kcp-dev/kcp#3350
-		// created by: k8s.io/client-go/util/workqueue.newDelayingQueue[...]
-		goleak.IgnoreTopFunction("k8s.io/client-go/util/workqueue.(*delayingType[...]).waitingLoop"),
-		// created by: k8s.io/client-go/util/workqueue.newQueue[...]
-		goleak.IgnoreTopFunction("k8s.io/client-go/util/workqueue.(*Typed[...]).updateUnfinishedWorkLoop"),
-		// created by: github.com/kcp-dev/apimachinery/v2/third_party/informers.newScopedSharedIndexInformerWithContext
-		goleak.IgnoreTopFunction("github.com/kcp-dev/apimachinery/v2/third_party/informers.newScopedSharedIndexInformerWithContext.func1"),
-
-		// unknown
-		// created by: net/http.(*Transport).dialConn
-		goleak.IgnoreTopFunction("net/http.(*persistConn).writeLoop"),
-		// created by: net/http.(*Server).Serve
-		goleak.IgnoreTopFunction("golang.org/x/net/http2.(*serverConn).serve"),
-		// created by: gopkg.in/natefinch/lumberjack%2ev2.(*Logger).mill.func1
-		goleak.IgnoreTopFunction("gopkg.in/natefinch/lumberjack%2ev2.(*Logger).millRun"),
-		// created by: golang.org/x/net/http2.(*serverConn).serve
-		goleak.IgnoreTopFunction("internal/poll.runtime_pollWait"),
-	}
-
-	// 14s as etcd sets a client request timeout of up to 7 seconds when
-	// shutting down the server and then starts shutting down everything
-	// else.
-	// A shorter timespan d would lead to false positives in the tests.
-	WaitTime = 14 * time.Second
-)
-
-// GoleakWithDefaults verifies that there are no goroutine leaks.
-// Goleak tests cannot be run in parallelized tests.
-func GoleakWithDefaults(tb testing.TB, in ...goleak.Option) {
-	tb.Helper()
-
-	tb.Logf("waiting %v for goroutines to finish", WaitTime)
-	time.Sleep(WaitTime)
-
-	opts := append(KnownGoroutineLeaks, in...)
-	goleak.VerifyNone(tb, opts...)
-}
```

Comment:
So the idea here was to use IgnoreCreatedBy to ignore any goroutines started by KCP when testing a component for leaks. However that requires the PR to be merged: https://github.com/uber-go/goleak/pull/136

Since that doesn't seem to happen anytime soon I removed the entire helper function. The leak test for KCP itself is in framework/leak_test for now, the workspace leak test in workspace/leak_test.